### DISCUSSION TITLE
IntoIntoDart error

### DISCUSSION THREAD

**Discussion by alex-cryptogaming** [rustdesk/rustdesk#5477] [General]
on start ./flutter/ndk_arm64.sh

error[E0432]: unresolved import `flutter_rust_bridge::rust2dart::IntoIntoDart`
  --> src/bridge_generated.rs:16:5
   |
16 | use flutter_rust_bridge::rust2dart::IntoIntoDart;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^------------
   |     |                               |
   |     |                               help: a similar name exists in the module: `IntoDart`
   |     no `IntoIntoDart` in `rust2dart`

error[E0277]: the trait bound `DartCObject: From<EventToUI>` is not satisfied
   --> src/flutter.rs:164:41
    |
164 |     pub event_stream: Arc<RwLock<Option<StreamSink<EventToUI>>>>,
    |                                         ^^^^^^^^^^^^^^^^^^^^^ the trait `From<EventToUI>` is not implemented for `DartCObject`
    |
    = help: the following other types implement trait `From<T>`:
              <DartCObject as From<DartOpaque>>
              <DartCObject as From<RustOpaque<T>>>
    = note: required for `EventToUI` to implement `Into<DartCObject>`
    = note: required for `EventToUI` to implement `IntoDart`
note: required by a bound in `StreamSink`
   --> /home/uadmin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge-1.75.3/src/rust2dart.rs:97:26
    |
97  | pub struct StreamSink<T: IntoDart> {
    |                          ^^^^^^^^ required by this bound in `StreamSink`

error[E0277]: the trait bound `DartCObject: From<EventToUI>` is not satisfied
   --> src/flutter.rs:790:19
    |
790 |     event_stream: StreamSink<EventToUI>,
    |                   ^^^^^^^^^^^^^^^^^^^^^ the trait `From<EventToUI>` is not implemented for `DartCObject`
    |
    = help: the following other types implement trait `From<T>`:
              <DartCObject as From<DartOpaque>>
              <DartCObject as From<RustOpaque<T>>>
    = note: required for `EventToUI` to implement `Into<DartCObject>`
    = note: required for `EventToUI` to implement `IntoDart`
note: required by a bound in `StreamSink`
   --> /home/uadmin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge-1.75.3/src/rust2dart.rs:97:26
    |
97  | pub struct StreamSink<T: IntoDart> {
    |                          ^^^^^^^^ required by this bound in `StreamSink`

error[E0277]: the trait bound `DartCObject: From<EventToUI>` is not satisfied
   --> src/flutter_ffi.rs:110:16
    |
110 |     events2ui: StreamSink<EventToUI>,
    |                ^^^^^^^^^^^^^^^^^^^^^ the trait `From<EventToUI>` is not implemented for `DartCObject`
    |
    = help: the following other types implement trait `From<T>`:
              <DartCObject as From<DartOpaque>>
              <DartCObject as From<RustOpaque<T>>>
    = note: required for `EventToUI` to implement `Into<DartCObject>`
    = note: required for `EventToUI` to implement `IntoDart`
note: required by a bound in `StreamSink`
   --> /home/uadmin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge-1.75.3/src/rust2dart.rs:97:26
    |
97  | pub struct StreamSink<T: IntoDart> {
    |                          ^^^^^^^^ required by this bound in `StreamSink`

error[E0277]: the trait bound `DartCObject: From<EventToUI>` is not satisfied
    --> src/flutter_ffi.rs:1616:61
     |
1616 | pub fn plugin_register_event_stream(_id: String, _event2ui: StreamSink<EventToUI>) {
     |                                                             ^^^^^^^^^^^^^^^^^^^^^ the trait `From<EventToUI>` is not implemented for `DartCObject`
     |
     = help: the following other types implement trait `From<T>`:
               <DartCObject as From<DartOpaque>>
               <DartCObject as From<RustOpaque<T>>>
     = note: required for `EventToUI` to implement `Into<DartCObject>`
     = note: required for `EventToUI` to implement `IntoDart`
note: required by a bound in `StreamSink`
    --> /home/uadmin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge-1.75.3/src/rust2dart.rs:97:26
     |
97   | pub struct StreamSink<T: IntoDart> {
     |                          ^^^^^^^^ required by this bound in `StreamSink`

Some errors have detailed explanations: E0277, E0432.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `rustdesk` (lib) due to 5 previous errors
note: If the build failed due to a missing target, you can run this command:
note: 
note:     rustup target install aarch64-linux-android

**Comment by willharwell** [comment]
I'm also seeing this exact same output.

**Reply by mbahadirerkan** [reply]
cargo install flutter_rust_bridge_codegen --version "1.75.3" --features "uuid" 

You would probably need this version in the cargo.lock file. You should make sure that it works with the correct version in the cargo.lock file.

**Reply by nearchan** [reply]
not working~

**Comment by willharwell** [comment]
error[E0432]: unresolved import flutter_rust_bridge::rust2dart::IntoIntoDart

Fixed with:
cargo update -p flutter_rust_bridge

On to the next error:
error[E0277]: the trait bound `EventToUI: IntoIntoDart<_>` is not satisfied

**Reply by mbahadirerkan** [reply]
have you found a solution to that?

**Reply by alex-cryptogaming** [reply]
no

**Comment by Zacorich** [comment]
To fix the issue follow the following environment configuration:
https://rustdesk.com/docs/en/dev/build/all/
https://github.com/rustdesk/rustdesk/discussions/5561
https://github.com/rustdesk/rustdesk/discussions/763
